### PR TITLE
API documentation guide

### PR DESCRIPTION
**Pull request by samthor**

### CODE REVIEW

File: site/en/docs/handbook/how-to/add-an-api/index.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+api: browserAction
+---
+
+Something about this API. This text is completely optional. If you don't want to write anything,
+then you should include a comment like <!-- This is intentionally empty -->.
+```
+
+The API name is assumed to be under "chrome.apiName". If you are documenting an API like
+"chrome.input.ime", then by convention, the folder name should be "input_ime" but the API in your
+frontmatter shold still be `api: input.ime`.
```
Comment (robdodson) [inline site/en/docs/handbook/how-to/add-an-api/index.md:25]:
```suggestion
frontmatter should still be `api: input.ime`.
```

File: site/en/docs/handbook/how-to/add-an-api/index.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+The existence of this page will cause the API itself to be displayed in the
+[top-level reference page][1]. If it's a Chrome Apps-only API, it will appear at the very bottom
+of that page, rather than in Stable, Beta or Dev.
+
+## API Source
+
+The API reference is generated via the published APIs in [chrome-types on NPM][2]. It's possible
```
Comment (robdodson) [inline site/en/docs/handbook/how-to/add-an-api/index.md:33]:
```suggestion
The API reference is generated via the published APIs in [chrome-types on npm][2]. It's possible
```

I think the npm folks lowercase it
image

File: site/en/docs/handbook/how-to/add-an-api/index.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- Run `npm install chrome-types` to ensure that latest version is available.
+- Run `npm run types` to build our fast internal reprensetation of the TypeScript definition files (this normally happens on checkout, but since you've updated the types, it needs a re-run)
+
+If you're just curious about what APIs are published (and not all are documented, matching the
+previous incanation of this site) then take a look inside `node_modules/chrome-types/index.d.ts`
+(or its peer file `platform_app.d.ts` for Chrome Apps-only APIs).
+
+### Updating APIs
+
+Our TypeScript definition files are generated by by running "npm deploy" from within
```
Comment (robdodson) [inline site/en/docs/handbook/how-to/add-an-api/index.md:45]:
```suggestion
Our TypeScript definition files are generated by running "npm deploy" from within
```

**Review by robdodson — APPROVED** [review]

**Review by Ciampi10 — COMMENTED** [review]
> @ @[claurau-patch-1]()

**Pull request merged**